rocks-db/src/bin/leaf_checker/main.rs

```diff
@@ -117,6 +94,88 @@ pub async fn main() {
     );
 }
 
+async fn process_batch(
+    storage: Arc<Storage>,
+    batch_data: &mut Vec<Vec<u8>>,
+    delete_batch: &mut Vec<(Pubkey, u64)>,
+) {
+    let mut tasks = JoinSet::new();
+
+    for data in batch_data.iter() {
+        tasks.spawn(process_leaf(storage.clone(), data.clone()));
+    }
+
+    while let Some(task_result) = tasks.join_next().await {
+        match task_result {
+            Ok(key) => {
+                if let Some(key) = key {
+                    delete_batch.push(key);
+                }
+            }
+            Err(err) if err.is_panic() => {
+                let err = err.into_panic();
+                println!("Task panic: {:?}", err);
+            }
+            Err(err) => {
+                let err = err.to_string();
+                println!("Task error: {}", err);
+            }
+        }
+    }
+
+    batch_data.clear();
+}
+
+async fn process_leaf(storage: Arc<Storage>, data: Vec<u8>) -> Option<(Pubkey, u64)> {
+    let cl_leaf_data = deserialize::<ClLeaf>(data.as_ref()).unwrap();
+
+    let key = ClItemKey::new(cl_leaf_data.cli_node_idx, cl_leaf_data.cli_tree_key);
+
+    let cl_item_rows = storage
+        .cl_items
+        .batch_get(vec![key])
```

Comment:
batch_get allows you to select a batch that'll return None if there is no value, no need to select individually. Also you're more interested in missing keys, rather then values, rocks has some methods to work with those

Reply:
thanks, done it.
it gave additional 6% boost